module/zfs/dmu_object.c

```diff
@@ -32,6 +32,15 @@
 #include <sys/zfeature.h>
 #include <sys/dsl_dataset.h>
 
+/*
+ * Each of the concurrent object allocators will grab
+ * 2^dmu_object_alloc_chunk_shift objects at a time.  The default is to
+ * grab 128 object, which s 4 blocks worth.  This was experimentally
+ * determined to be the lowest value that eliminates the measurable effect
+ * of lock contention from this code path.
+ */
+int dmu_object_alloc_chunk_shift = 7;
```

Comment:
The tunable is expressed in the number of objects (technically number of slots).

I assume you mean that I should say "dnode slots" rather than "dnodes".